sdk/batch/batch-rest/MigrationGuide.md

```diff
@@ -0,0 +1,265 @@
+
+```typescript
+import { BatchServiceClient } from "@azure/batch";
+import { loginWithVmMSI } from "@azure/ms-rest-nodeauth";
+
+const credential = await loginWithVmMSI({
+  resource: "https://batch.core.windows.net/"
+});
+const client = new BatchServiceClient(credential, "<account-endpoint>");
+```
+
+Now in `@azure-rest/batch`, you can pass any of the [credentials from the `@azure/identity` package](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/samples/AzureIdentityExamples.md) to the `createClient` method to make use of your Microsft Entra ID credentials. In following sample, it creates an instance of [`DefaultAzureCredential`](https://learn.microsoft.com/javascript/api/@azure/identity/defaultazurecredential) to authenticate with the Azure Batch service.
+
+```typescript
+import { DefaultAzureCredential } from "@azure/identity";
+import createClient from "@azure-rest/batch";
+
+const credential = new DefaultAzureCredential();
+const client = createClient("<account-endpoint>", credential);
+```
+
+## Operation response differences
+
+Previously in `@azure/batch`, the client operation return a `Promise` that resolves to the result of the response body JSON. The following example demonstrates how to get a job with the `BatchServiceClient` instance.
+
+```typescript
+const job = await client.job.get("<job-id>");
+console.log(`Job id: ${job.id}, state: ${job.state}`);
+```
+
+Now in `@azure-rest/batch`, the client operation return a `Promise` that resolves to the response object, which contains the response body and the status code. In order to get the response body JSON, you need to first check if the response is unexpected with the `isUnexpected` helper method, then access the response body. The following example demonstrates how to get a job in `@azure-rest/batch`.
+
+```typescript
+import { isUnexpected } from '@azure-rest/batch';
+const response = await client.path("/jobs/{jobId}", "<job-id>").get();
+if (isUnexpected(response)) {
+  throw new Error(`Failed to get job: ${response.body.message}`);
+}
+console.log(`Response status code: ${response.status}`);
+
+const job = response.body;
+console.log(`Job id: ${job.id}, state: ${job.state}`);
+```
+
+## Error handling
+
+Previously in `@azure/batch`, the client operation succeed only when service return expected HTTP status code, for example `201` for create resources operations or `200` for general HTTP GET requests. Unexpected HTTP status code will throw a `RestError` from `@azure/ms-rest-js` package. The following example demonstrate how to handle different errors might occur in the get pool request.
```

Comment:
Suggestion:

Previously in `@azure/batch`, client operations succeeded only when the service returned an expected HTTP status code (usually 200 or 201). Unexpected status codes would throw a `RestError` from `@azure/ms-rest-js` package. The following example demonstrates how to handle different errors which might occur when getting a pool.